lisa/_kmod.py

```diff
@@ -2267,4 +2286,40 @@ def from_target(cls, target, **kwargs):
     def _event_features(cls, events):
         return set(f'event__{event}' for event in events)
 
+    def install(self, kmod_params=None):
+        try:
+            self.uninstall()
+        except Exception:
+            pass
+
+        busybox = quote(self.target.busybox)
+        modules_path_base = '/lib/modules'
+        modules_version = self.target.kernel_version.release
+
+        if self.target.os == 'android':
+            modules_path_base = f'/vendor_dlkm{modules_path_base}'
+            try:
+                modules_version = self.target.execute(
+                    f"{busybox} ls {modules_path_base} | {busybox} head -1"
+                ).strip()
+            except TargetStableCalledProcessError:
+                pass
+
+        modules_path = f"{modules_path_base}/{modules_version}"
+
+        try:
+            lisa_module_path = self.target.execute(
+                f"{busybox} find {modules_path} -name '{self.src.mod_name}.ko'"
+            ).strip()
+
+            self.target.execute(
+                f"{busybox} insmod {lisa_module_path} version={self.src.checksum_sources}"
+            )
+        except TargetStableCalledProcessError as e:
+            if e.returncode == errno.EPROTO:
+                raise ValueError('In-tree module version does not match the Lisa checkout.')
```

Comment:
nit: s/Lisa checkout/what lisa expects/

LISA does not necessarily come from a git checkout, it can be installed from PyPI